File: 2nd Year/1st Semester/Metode Avansate de Programare/Sesiune/ApeleRomane/ApeleRomane/src/main/java/org/example/apeleromane/Repository/IRepository.java
package org.example.apeleromane.Repository;


import org.example.apeleromane.Domain.Entity;

import java.util.List;

public interface IRepository<ID, E extends Entity<ID>> {
    E findOne(ID id);

    List<E> findAll();

    E save(E entity);

    E delete(ID id);

    E update(E entity);
}

File: 2nd Year/2nd Semester/MPP/Triatlon-C#/TriatlonPersistance/IRepository.cs
using TriatlonModel;

namespace TriatlonPersistance
{
    public interface IRepository<ID, E> where E : Entity<ID>
    {
        E FindOne(ID id);
        List<E> FindAll();
        void Save(E entity);
        void Delete(ID id);
        void Update(E entity);
    }
}